website/project/model.py

```diff
@@ -1411,6 +1411,35 @@ def set_node_license(self, license_id, year, copyright_holders, auth, save=False
         if save:
             self.save()
 
+    def subscribe_user_to_notifications(self, user):
+        """ Update the notification settings for the creator or contributors
+
+        :param user: User to subscribe to notifications
+        """
+        from website.notifications.utils import to_subscription_key
+        from website.notifications.utils import get_global_notification_type
+        from website.notifications.model import NotificationSubscription
+        from website.notifications.constants import NODE_SUBSCRIPTIONS_AVAILABLE as events
+
+        notification_type = 'email_transactional'
+        target_id = self._id
+
+        if user.is_registered:
+            for event in events:
+                event_id = to_subscription_key(target_id, event)
+                global_event_id = to_subscription_key(user._id, 'global_' + event)
+                global_subscription = NotificationSubscription.load(global_event_id)
+
+                subscription = NotificationSubscription.load(event_id)
+                if not subscription:
+                    subscription = NotificationSubscription(_id=event_id, owner=self, event_name=event)
+                if global_subscription:
+                    global_notification_type = get_global_notification_type(global_subscription, user)
+                    subscription.add_user_to_subscription(user, global_notification_type)
```

Comment:
...unless `add_user_to_subscription` is responsible for handling `none`, which on a quick glance, it does.